File: app/src/main/java/com/github/malitsplus/shizurunotes/data/action/ChangeEnergyRecoveryRatioByDamageAction.java
package com.github.malitsplus.shizurunotes.data.action;

import com.github.malitsplus.shizurunotes.R;
import com.github.malitsplus.shizurunotes.common.I18N;
import com.github.malitsplus.shizurunotes.data.Property;
import com.github.malitsplus.shizurunotes.data.Skill;

public class ChangeEnergyRecoveryRatioByDamageAction extends ActionParameter {

    @Override
    protected void childInit() {
        super.childInit();
    }

    protected String getChildrenActionString() {
        StringBuilder childrenActionString = new StringBuilder();
        if (childrenAction != null) {
            for (Skill.Action action : childrenAction) {
                childrenActionString.append(action.getActionId() % 10).append(", ");
            }
            childrenActionString.delete(childrenActionString.lastIndexOf(", "), childrenActionString.length());
        }
        return childrenActionString.toString();
    }

    @Override
    public String localizedDetail(int level, Property property) {
        return I18N.getString(R.string.change_energy_recovery_ratio_of_action_s1_to_s2_when_s3_get_damage,
                getChildrenActionString(),
                actionValue1.valueString(),
                targetParameter.buildTargetClause());
    }
}

File: app/src/main/java/com/github/nyanfantasia/shizurunotes/data/action/ChangeEnergyRecoveryRatioByDamageAction.kt
package com.github.nyanfantasia.shizurunotes.data.action

import com.github.nyanfantasia.shizurunotes.R
import com.github.nyanfantasia.shizurunotes.common.I18N.Companion.getString
import com.github.nyanfantasia.shizurunotes.data.Property

class ChangeEnergyRecoveryRatioByDamageAction : ActionParameter() {

    private val childrenActionString: String
        get() {
            val childrenActionString = StringBuilder()
            if (childrenAction != null) {
                for (action in childrenAction!!) {
                    childrenActionString.append(action.actionId % 10).append(", ")
                }
                childrenActionString.delete(
                    childrenActionString.lastIndexOf(", "),
                    childrenActionString.length
                )
            }
            return childrenActionString.toString()
        }

    override fun localizedDetail(level: Int, property: Property): String {
        return getString(
            R.string.change_energy_recovery_ratio_of_action_s1_to_s2_when_s3_get_damage,
            childrenActionString,
            actionValue1.valueString(),
            targetParameter.buildTargetClause()
        )
    }
}